Utils.py

```diff
@@ -166,6 +166,9 @@ def home_path(*path: str) -> str:
                 os.symlink(home_path.cached_path, legacy_home_path)
             else:
                 os.makedirs(home_path.cached_path, 0o700, exist_ok=True)
+    elif sys.platform == 'darwin':
+        home_path.cached_path = os.path.expanduser('~/Library/Application Support/Archipelago')
```

Comment:
2 things:
* Is this path ("Application Support") the correct one? It seems a bit odd.
  If yes, can you use platformdirs.user_data_dir("Archipelago") instead?
  (We should probably change the Linux code path to platformdirs as well)
* Are macos users fine with **not** upgrading the paths? I.e. in the Linux code path, we move files from the old location to the new location, so the roms, settings in host.yaml, etc. don't disappear when upgrading.

Reply:
1. Yep, that is the correct location.  And yeah, I can use platformdirs instead.

2. The way macos users upgrade, you're deleting the entire app bundle and installing the new version fresh.  There's not really anything left to migrate.